src/nccl_ofi_net.c

```diff
@@ -1443,11 +1466,19 @@ static ncclResult_t ofi_connect(int dev, void *handle, void **sendComm)
 	} while (true);
 
 	/* Ensure the message is sent. */
-	do {
-		ret = nccl_ofi_progress(nccl_ofi_component[dev]);
-		if (OFI_UNLIKELY(ret != 0))
-			goto error;
-	} while (req->state != NCCL_OFI_REQ_COMPLETED);
+	/* Skip this step for sending to self */
+	if (connect_to_self) {
+		/* prevent req from freeing */
+		req = NULL;
```

Comment:
I would rather prefer checking value of `connect_to_self` when we actually free `req` rather than making it NULL here.

Reply:
ok

Reply:
done